internal/proxymanager/config.go

```diff
@@ -0,0 +1,19 @@
+package proxymanager
+
+// Config is the exported proxy configuration type for users of the proxymanager API.
+type Config struct {
+	HTTPProxy  string
+	HTTPSProxy string
+	FTPProxy   string
+	SocksProxy string
+	NoProxy    string
```

Comment:
I got slapped by the `clippy` tool in Rust when I set up a Struct like this as well. It told me something like `if you are (post | pre) fixing every field in here with the same word, there's probably a better way to write it`.
Even though this is not Rust, I quite liked the suggestion and maybe we can figure something out here?